commands/src/test/java/discord4j/commands/example/LogoutCommand.java

```diff
@@ -0,0 +1,28 @@
+package discord4j.commands.example;
+
+import discord4j.commands.BaseCommand;
+import discord4j.commands.exceptions.CommandException;
+import discord4j.core.event.domain.message.MessageCreateEvent;
+import reactor.core.publisher.Mono;
+
+import java.util.Optional;
+
+public class LogoutCommand implements BaseCommand {
+
+    @Override
+    public Mono<Void> execute(MessageCreateEvent event) {
+        return event.getClient()
+                    .getApplicationInfo()
+                    .filter(info -> info.getOwnerId().equals(event.getMessage().getAuthorId().get()))
+                    .switchIfEmpty(Mono.fromRunnable(() -> {
```

Comment:
I think the verbosity here may be hinting that using exceptions for this kind of thing isn't great. Even with a more reasonable CommandException constructor, we have `switchIfEmpty(Mono.fromRunnable() -> { throw ...` just to express a simple permissions failure.

Reply:
Generally, though, this logic would probably be centralized in a "checkPermissions" method made by the user. These examples could probably do with a rewrite as they are not exactly well designed for end user consumption

Reply:
The examples should be redone before merge.